tooling/internal/matlab/types/types.go

```diff
@@ -0,0 +1,1099 @@
+		}
+	}
+
+	return nil
+}
+
+func writeUnionClasses(fw *common.MatlabFileWriter, td dsl.TypeDefinition, unionGenerated map[string]bool) error {
+	var writeError error
+	dsl.Visit(td, func(self dsl.Visitor, node dsl.Node) {
+		switch node := node.(type) {
+		case *dsl.GeneralizedType:
+			if node.Cases.IsUnion() {
+				unionClassName := common.UnionClassName(node)
+				if !unionGenerated[unionClassName] {
+					if _, isNamedType := td.(*dsl.NamedType); isNamedType {
+						// This is a named type defining a union, so we will use the named type's name instead
+						unionClassName = td.GetDefinitionMeta().Name
+					}
+					writeError = fw.WriteFile(unionClassName, func(w *formatting.IndentedWriter) {
+						writeUnionClass(w, unionClassName, node, td.GetDefinitionMeta().Namespace)
+					})
+					if writeError != nil {
+						return
+					}
+					unionGenerated[unionClassName] = true
+				}
+			}
+		}
+		self.VisitChildren(node)
+	})
+
+	return writeError
+}
+
+func writeUnionClass(w *formatting.IndentedWriter, className string, generalizedType *dsl.GeneralizedType, contextNamespace string) error {
+	qualifiedClassName := fmt.Sprintf("%s.%s", common.NamespaceIdentifierName(contextNamespace), className)
+	fmt.Fprintf(w, "classdef %s < yardl.Union\n", className)
+	common.WriteBlockBody(w, func() {
+		w.WriteStringln("methods (Static)")
+		common.WriteBlockBody(w, func() {
+			index := 1
+			for _, tc := range generalizedType.Cases {
+				if tc.Type == nil {
+					continue
+				}
+
+				fmt.Fprintf(w, "function res = %s(value)\n", formatting.ToPascalCase(tc.Tag))
+				common.WriteBlockBody(w, func() {
+					fmt.Fprintf(w, "res = %s(%d, value);\n", qualifiedClassName, index)
+				})
+				index += 1
+				w.WriteStringln("")
+			}
+
+			writeZerosStaticMethod(w, qualifiedClassName, []string{"0", "yardl.None"})
+		})
+		w.WriteStringln("")
+
+		w.WriteStringln("methods")
+		common.WriteBlockBody(w, func() {
+			index := 1
+			var tags []string
+			for _, tc := range generalizedType.Cases {
+				if tc.Type == nil {
+					continue
+				}
+				tag := formatting.ToPascalCase(tc.Tag)
+				tags = append(tags, fmt.Sprintf(`"%s"`, tag))
+				fmt.Fprintf(w, "function res = is%s(self)\n", tag)
+				common.WriteBlockBody(w, func() {
+					fmt.Fprintf(w, "res = self.index == %d;\n", index)
+				})
+				index += 1
+				w.WriteStringln("")
+			}
+
+			w.WriteStringln("function eq = eq(self, other)")
+			common.WriteBlockBody(w, func() {
+				fmt.Fprintf(w, "eq = isa(other, \"%s\") && isequal(self.index, other.index) && isequal(self.value, other.value);\n", qualifiedClassName)
+			})
+			w.WriteStringln("")
+			w.WriteStringln("function ne = ne(self, other)")
+			common.WriteBlockBody(w, func() {
+				w.WriteStringln("ne = ~self.eq(other);")
+			})
+			w.WriteStringln("")
+			w.WriteStringln("function t = tag(self)")
+			common.WriteBlockBody(w, func() {
+				fmt.Fprintf(w, "tags_ = [%s];\n", strings.Join(tags, ", "))
+				w.WriteStringln("t = tags_(self.index_);")
```

Comment:
Can we easily add a string representation of the value?

Reply:
My last change added both the Union `tag` *and* the `value` to all Union string representations. When MATLAB displays an object property that is not a built-in type (e.g. all yardl-generated classes), it summarizes the property as a `[1x1 <classname>]`.

The following code
```matlab
i = 42;
disp(i);
u = test_model.AliasedIntOrSimpleRecord.Int32(i);
disp(u);
```
prints:
```
    42

AliasedIntOrSimpleRecord.Int32

    value: 42
````

and the following code
```matlab
sr = test_model.SimpleRecord(x=1, y=2, z=3);
disp(sr);
u = test_model.AliasedIntOrSimpleRecord.SimpleRecord(sr);
disp(u);
```
prints:
```
  SimpleRecord with properties:

    x: 1
    y: 2
    z: 3

AliasedIntOrSimpleRecord.SimpleRecord

    value: [1x1 test_model.SimpleRecord]
```